src/ledger/OfferFrame.cpp

```diff
@@ -120,7 +120,16 @@ OfferFrame::getFlags() const
 bool
 OfferFrame::isValid(OfferEntry const& oe)
 {
-    return isAssetValid(oe.buying) && isAssetValid(oe.selling);
+    bool res = isAssetValid(oe.buying);
+    // to avoid SQL problems (not a protocol limitation)
+    res = res && (oe.offerID <= INT64_MAX);
+
+    res = res && (oe.amount > 0);
+    res = res && (oe.price.n > 0);
+    res = res && (oe.price.d >= 1);
+    res = res && isAssetValid(oe.selling);
```

Comment:
It might be neater (and less brainpower to maintain in the future) to keep both isAssetValid checks together, since they are analogous. This is purely a question of style though

Reply:
fixed